lib/metasploit/framework/login_scanner/kerberos.rb

```diff
@@ -0,0 +1,93 @@
+            # Stop further requests entirely
+            res_opts = request_options.merge({ status: :eof, proof: e })
+            return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+          rescue Rex::Proto::Kerberos::Model::Error::KerberosDecodingError => e
+            elog(e)
+            # Stop further requests entirely
+            res_opts = request_options.merge({ status: :decode_error, proof: e })
+            return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+          end
+
+          case res.msg_type
+          when Rex::Proto::Kerberos::Model::AS_REP
+            hash = format_asrep_to_john_hash(res)
+
+            # Accounts that have 'Do not require Kerberos preauthentication' enabled, will receive an ASREP response with a ticket present
+            res_opts = request_options.merge({ status: :no_preauth, hash: hash, proof: res.msg_type })
+            return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+          when Rex::Proto::Kerberos::Model::KRB_ERROR
+            if res.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KDC_ERR_PREAUTH_REQUIRED
+              res_opts = request_options.merge({ status: :present, proof: res.msg_type })
+              return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+            elsif res.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KDC_ERR_CLIENT_REVOKED
+              res_opts = request_options.merge({ status: :disabled_or_locked_out, proof: res.msg_type })
+              return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+            elsif res.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KDC_ERR_C_PRINCIPAL_UNKNOWN
+              res_opts = request_options.merge({ status: :not_found, proof: res.msg_type })
+              return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+            elsif res.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KDC_ERR_WRONG_REALM
+              # Stop further requests entirely
+              res_opts = request_options.merge({ status: :wrong_realm, proof: res.msg_type })
+              return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+            else
+              res_opts = request_options.merge({ status: :unknown_error, proof: res.error_code })
+              return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+            end
+          else
+            res_opts = request_options.merge({ status: :unknown_response, proof: { error_code: res.error_code, response: res.msg_type.inspect } })
+            return Metasploit::Framework::LoginScanner::Result.new(res_opts)
+          end
+        end
+
+        def scan!
+          cred_details.each do |credential|
+            result = attempt_login(credential)
+            result.freeze
+
+            yield result if block_given?
+          end
+        end
+
+        attr_accessor :server_name, :realm, :pa_data, :datastore
```

Comment:
```suggestion
        attr_accessor :server_name, :pa_data, :datastore
```

As before - this would be on the credential itself 👀